app/assets/javascripts/models/story.js

```diff
@@ -62,55 +62,31 @@ var Story = module.exports = Backbone.Model.extend({
     model.setColumn();
   },
 
-  moveBetween: function(before, after) {
-    var beforeStory = this.collection.get(before);
-    var afterStory = this.collection.get(after);
-    var difference = (afterStory.position() - beforeStory.position()) / 2;
-    var newPosition = difference + beforeStory.position();
-    this.set({position: newPosition});
-    this.collection.sort();
+  move: function(previous_story_id, next_story_id) {
+    var newPosition = this.collection.calculateNewPosition(previous_story_id, next_story_id)
+    this.setPosition(newPosition);
     return this;
   },
 
-  moveAfter: function(beforeId) {
-    var before = this.collection.get(beforeId);
-    var after = this.collection.nextOnColumn(before);
-    var afterPosition;
-    if (typeof after === 'undefined') {
-      afterPosition = before.position() + 2;
-    } else {
-      afterPosition = after.position();
+  setPosition: function(position) {
+    this.set({position: position});
```

Comment:
Just a tip (not needed to change if you think we don't need to): since we're using ES6 here, you can simplify this with `this.set({ position });` it'll use the local variable with the same name of the key.